privacy-on-beam/pbeam/sum_test.go

```diff
@@ -340,6 +475,40 @@ func TestSumPerKeyCrossPartitionContributionBoundingFloat(t *testing.T) {
 	}
 }
 
+// Checks that SumPerKey with partitions bounds per-user contributions correctly with float values.
+// The logic mirrors TestCountCrossPartitionContributionBounding.
+func TestSumPerKeyWithPartitionsCrossPartitionContributionBoundingFloat(t *testing.T) {
+	// triples contains {1,0,1.0}, {2,0,1.0}, …, {50,0,1.0}, {1,1,1.0}, …, {50,1,1.0}, {1,2,1.0}, …, {50,9,1.0}.
+	var triples []tripleWithFloatValue
+	for i := 0; i < 10; i++ {
+		triples = append(triples, makeDummyTripleWithFloatValue(50, i)...)
+	}
+	result := []testFloat64Metric{
+		{0, 150.0},
+	}
+	p, s, col, want := ptest.CreateList2(triples, result)
+	col = beam.ParDo(s, extractIDFromTripleWithFloatValue, col)
+	partitions := []int{0, 1, 2}
```

Comment:
We are not actually testing cross-partition contribution bounding if we have a total of 3 kept partitions. We should keep more partitions and let Privacy on Beam drop the extra ones. So, let's say you keep `{0, 1, 2, 3, 4}`.

Reply:
Same as previous comment. Testing that bound is not wasted on unspecified partitions.